packages/core/admin/admin/src/content-manager/components/DynamicZone/components/DynamicComponent.js

```diff
@@ -0,0 +1,198 @@
+import React, { useMemo, useState } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { useIntl } from 'react-intl';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import get from 'lodash/get';
+
+import { Accordion, AccordionToggle, AccordionContent } from '@strapi/design-system/Accordion';
+import { IconButton } from '@strapi/design-system/IconButton';
+import { Box } from '@strapi/design-system/Box';
+import { Flex } from '@strapi/design-system/Flex';
+import { Stack } from '@strapi/design-system/Stack';
+
+import { useCMEditViewDataManager } from '@strapi/helper-plugin';
+
+import Trash from '@strapi/icons/Trash';
+import ArrowDown from '@strapi/icons/ArrowDown';
+import ArrowUp from '@strapi/icons/ArrowUp';
+
+import { useContentTypeLayout } from '../../../hooks';
+import { getTrad } from '../../../utils';
+
+import FieldComponent from '../../FieldComponent';
+
+const ActionStack = styled(Stack)`
+  svg {
+    path {
+      fill: ${({ theme, expanded }) =>
+        expanded ? theme.colors.primary600 : theme.colors.neutral600};
+    }
+  }
+`;
+
+const IconButtonCustom = styled(IconButton)`
+  background-color: transparent;
+`;
+
+const StyledBox = styled(Box)`
+  > div:first-child {
+    box-shadow: ${({ theme }) => theme.shadows.tableShadow};
+  }
+`;
+
+const AccordionContentRadius = styled(Box)`
+  border-radius: 0 0 ${({ theme }) => theme.spaces[1]} ${({ theme }) => theme.spaces[1]};
+`;
+
+const Rectangle = styled(Box)`
+  width: ${({ theme }) => theme.spaces[2]};
+  height: ${({ theme }) => theme.spaces[4]};
+`;
+
+const DynamicZoneComponent = ({
+  componentUid,
+  formErrors,
+  index,
+  isFieldAllowed,
+  onMoveComponentDownClick,
+  onMoveComponentUpClick,
+  name,
+  onRemoveComponentClick,
+  showDownIcon,
+  showUpIcon,
+}) => {
+  const [isOpen, setIsOpen] = useState(true);
+  const { formatMessage } = useIntl();
+  const { getComponentLayout } = useContentTypeLayout();
+  const { modifiedData } = useCMEditViewDataManager();
+  const { icon, friendlyName, mainValue } = useMemo(() => {
+    const componentLayoutData = getComponentLayout(componentUid);
+
+    const {
+      info: { icon, displayName },
+    } = componentLayoutData;
+
+    const mainField = get(componentLayoutData, ['settings', 'mainField'], 'id');
+
+    const displayedValue =
+      mainField === 'id' ? '' : toString(get(modifiedData, [name, index, mainField], ''));
+
+    const mainValue = displayedValue.trim().length < 1 ? '' : ` - ${displayedValue}`;
```

Comment:
I think there's an issue with the mainValue here 👀

<img width="1112" alt="image" src="https://user-images.githubusercontent.com/71838159/201687357-9f3d34f1-4b56-452f-a6eb-db1ac48cab6b.png">

didn't check the tests yet to see if it should caught it

Reply:
That might have been introduced by https://github.com/strapi/strapi/pull/12500 @joshuaellis do you want me to have a look? I think this might not be related to this PR.

Reply:
Can do if you want 😄  what was the point of this addition? I struggled to follow the conversation.

Reply:
The goal was to display the value of the main field as part of the accordion header, so that it is easier for users to differentiate between components. It should only display main fields which are not the ID field.